posts/shadow-dom-and-event-propagation.md

```diff
@@ -0,0 +1,267 @@
+  const tableValue = field => Array.from(el.querySelectorAll('th')).find(th => th.textContent.trim() === field)?.nextSibling.textContent.trim();
+  return { type: tableValue('Type'), bubbles: tableValue('Bubbles'), composed: tableValue('Composed') }
+});
+
+- https://w3c.github.io/touch-events/#touch-interface
+- https://w3c.github.io/pointerevents/#firing-events-using-the-pointerevent-interface
+- https://w3c.github.io/clipboard-apis/#clipboard-event-definitions
+-->
+
+Most of the standard [UI events](https://w3c.github.io/uievents) bubble and are composed with a few exceptions. There are a few exceptions like `mouseenter` and `mouseleave` that aren't bubbling and composed. Finally, the `slotchange` event stands out as the only bubbling and non-composed event.
+
+<details>
+    <summary>Complete event list</summary>
+
+- **Bubbling and non-composed events:** `slotchange`
+- **Non-bubbling and non-composed events:** `mouseenter`, `mouseleave`, `pointerenter`, `pointerleave`
+- **Bubbling and composed events:** `focusin`, `focusout`, `auxclick`, `click`, `dblclick`, `mousedown`, `mousemove`, `mouseout`, `mouseover`, `mouseup`, `wheel`, `input`, `keydown`, `keyup`, `keypress`, `touchstart`, `touchend`, `touchmove`, `pointerover`, `pointerdown`, `pointermove`, `pointerup`, `pointerout`
+
+</details>
+
+## `Event.prototype.target` and event retargeting
+
+The [`Event.prototype.target`](https://developer.mozilla.org/en-US/docs/Web/API/Event/target) property references the object from which the event was dispatched. When an event is dispatched from a DOM node, its `target` is set to the node from which the event originates.
+
+To preserve shadow DOM encapsulation and avoid leaking component internals, the **`target` is updated to the host element as events cross-shadow boundaries**. This process is called event retargeting. Let's take the same nested shadow tree example that we used before to illustrate this aspect:
+
+<event-visualizer label="Nested shadow trees" event-bubbles event-composed>
+  <template>
+    <div id="a">
+      <template shadowroot="open">
+        <div id="c">
+          <template shadowroot="open">
+            <span id="e" target></span>
+          </template>
+          <p id="d"></p>
+        </div>
+      </template>
+      <img id="b"></img>
+    </div>
+  </template>
+</event-visualizer>
+
+When the composed event propagates from `div#c` shadow root to the host element, the event `target` is set to `div#c`. In the same way, once the event propagates through `div#a` shadow boundary, the event target is set to `div#a`.
+
+One interesting side-effect of event retargeting is that once the event is done propagating the event `target` is always set to the outermost host element. In the case you are doing debouncing, the event `target` should be [cached](https://github.com/salesforce/lwc/issues/2265) otherwise the debounced method loses track of the event target.
```

Comment:
```suggestion
One interesting side-effect of event retargeting is that once the event is done propagating, the event `target` is always set to the outermost host element. In the case where you are doing debouncing, the event `target` should be [cached](https://github.com/salesforce/lwc/issues/2265), otherwise the debounced method loses track of the event target.
```